src/Test/AbstractIntegrationTestCase.php

```diff
@@ -269,14 +262,27 @@ function (FixerInterface $fixer) {
                 new NullCacheManager()
             );
 
+            Tokens::clearCache();
             $runner->fix();
             $fixedInputCodeWithReversedFixers = file_get_contents($tmpFile);
 
-            $this->assertNotSame(
-                $fixedInputCode,
-                $fixedInputCodeWithReversedFixers,
-                sprintf('Set priorities must be significant. If fixers used in reverse order return same output then the integration test is not sufficient or the priority relation between used fixers should not be set. In "%s".', $case->getFileName())
-            );
+            // If output is different depends on rules order - we need to verify that the rules are ordered by priority.
+            // If not, any order is valid.
+            if ($fixedInputCode !== $fixedInputCodeWithReversedFixers) {
+                $this->assertGreaterThan(
```

Comment:
wrong basing you made your conclusion on.

main `Runner` class perform fixing using rules
`FixerFactory` convert json describing rules one want to use for `FixerInterface[]` instances and **sort** them

IntegrationTest will take `fixers` from `FixerFactory` - that is already sorted.
Then, we run `Runner` with `fixers` in proper order to generate `fixedInputCode`.
Then, we run `Runner` with `fixers` in reverse order to generate `fixedInputCodeWithReversedFixers`